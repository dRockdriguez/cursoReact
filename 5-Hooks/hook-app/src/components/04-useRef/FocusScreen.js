import React, { useRef } from "react";
import "./focusScreen.css";

export const FocusScreen = () => {
  const inputRef = useRef();

  const handleFocus = () => {
    inputRef.current.select();
  };

  return (
    <>
      <h1>FocusScreen</h1>
      <hr />

      <input
        ref={inputRef}
        type="text"
        className="form-control"
        placeholder="Nombre..."
      />
      <button className="btn btn-outline-primary mt-3" onClick={handleFocus}>
        Focus
      </button>
    </>
  );
};
