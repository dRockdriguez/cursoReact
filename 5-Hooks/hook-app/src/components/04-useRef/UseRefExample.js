import React, { useState } from "react";
import "./focusScreen.css";
import { MultipleCustomHooks } from "../examples/MultipleCustomHooks";

export const UseRefExample = () => {
  const [show, setShow] = useState(true);

  return (
    <>
      <h1>UseRefExample</h1>
      <hr />

      {show && <MultipleCustomHooks />}

      <button
        className="btn btn-primary mt-8"
        onClick={() => {
          setShow(!show);
        }}
      >
        Toggle
      </button>
    </>
  );
};
