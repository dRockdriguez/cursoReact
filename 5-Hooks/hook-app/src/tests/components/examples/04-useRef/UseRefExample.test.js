import React from "react";
import { shallow } from "enzyme";
import { UseRefExample } from '../../../../components/04-useRef/UseRefExample';

describe("UseRefExample", () => {

    test("Debe mostrarse correctamente", () => {
        const wrapper = shallow(<UseRefExample />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("MultipleCustomHooks").exists()).toEqual(true);

    });

    test("Debe mostrar MultipleCustomHooks cuando se hace click", () => {
        const wrapper = shallow(<UseRefExample />);

        wrapper.find('button').simulate('click');
        expect(wrapper.find("MultipleCustomHooks").exists()).toEqual(false);

    });
});