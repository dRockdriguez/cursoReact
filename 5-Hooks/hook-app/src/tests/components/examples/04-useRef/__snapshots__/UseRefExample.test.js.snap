// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UseRefExample Debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    UseRefExample
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-primary mt-8"
    onClick={[Function]}
  >
    Toggle
  </button>
</Fragment>
`;
